### PR TITLE
Fixes reading HashiCorp Vault secrets

### PR DESCRIPTION
**Pull request by pavelgk**
## Contributor Checklist:

* [x] I have updated the unit tests
* [x]  I have created a file in the `master/buildbot/newsfragments` directory (and read the `README.txt` in that directory)

* [ ] I have updated the appropriate documentation

### CODE REVIEW

File: master/buildbot/test/unit/test_secret_in_vault.py [hunk 1]
```diff
@@ -110,23 +117,30 @@ def setUp(self):
...
         value = yield self.srvcVault.get("value")
         self.assertEqual(value, "value1")
 
+    @defer.inlineCallbacks
+    def testGetAnyKey(self):
+        self._http.expect(method='get', ep='/v1/secret/data/', params=None,
+                          data=None, json=None, code=200,
+                          content_json={"data": {"data": {"any_key": "value1"}}})
+        value = yield self.srvcVault.get("any_key")
+        self.assertEqual(value, "value1")
```
Comment (tardyp) [inline master/buildbot/test/unit/test_secret_in_vault.py:132]: How would I get this value from a Renderable object? This is not very clear for me, we need a test for it, and then documentation

**Review by tardyp — CHANGES_REQUESTED** [review]
Thanks for this fix!

**Review by p12tic — APPROVED** [review]
Looks good to me, thanks a lot!

**Comment by p12tic** [comment]
@pavelgk: Could you squash the commits to 1-2 commits so that the history is not polluted. Thanks!

**Comment by pavelgk** [comment]
@p12tic : Sure. Should I update documentation? It looks fine to me.

**Comment by p12tic** [comment]
Well, if the current documentation is not buggy it can stay as is :-)

File: master/buildbot/test/unit/test_secret_in_vault.py (hunk 1 above)
Comment (p12tic) [reply master/buildbot/test/unit/test_secret_in_vault.py:146]: Fixed now.

**Comment by p12tic** [comment]
So I've looked into adding a simple test covering the single line not covered in this PR and it turned out that the implementation is not fully correct as it breaks access by `key` identifier when the attribute name is not specified. I've fixed this and updated the PR.

Documentation on secret identifiers has been also added.

**Comment by pavelgk** [comment]
@p12tic I've prepared documentation with screenshots about how to use vault. Is it possible to add it?

**Comment by p12tic** [comment]
Oh, that's great news. Could you maybe create a separate PR for that?

**Comment by pavelgk** [comment]
@p12tic could you review my PR #5569?

**Pull request merged**